### PR TITLE
Paas newcmazpaasweb refactor

### LINKED ISSUE #18: Paas New-CmAzPaasWeb refactor - Changed settings file

### PR DESCRIPTION
**Pull request by deepend-dev**
Closes #18

### CODE REVIEW

File: Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1 [hunk 1]
```diff
@@ -43,15 +43,8 @@ function New-CmAzPaasWeb {
...
-			$resourceGroup = Get-CmAzService -Service $SettingsObject.ResourceGroupTag -Region $SettingsObject.Location -IsResourceGroup -ErrorAction SilentlyContinue
-			if (!$resourceGroup) {
-				$env:nameResourceGroup = Get-CmAzResourceName -Resource "ResourceGroup" -Architecture "PaaS" -Region $SettingsObject.location -Name $SettingsObject.resourceGroupName
-				Write-Verbose "Resourcegroup with tag $($SettingsObject.ResourceGroupTag) not found. New Resource Group '$($env:nameResourceGroup)' will be created."
-				New-AzResourceGroup -ResourceGroupName $env:nameResourceGroup -Location $SettingsObject.location -Force
-			}
-			else {
-				$env:nameResourceGroup = $resourceGroup.ResourceGroupName
-			}
+			$env:nameResourceGroup = Get-CmAzResourceName -Resource "ResourceGroup" -Architecture "PaaS" -Region $SettingsObject.location -Name $SettingsObject.resourceGroupName
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1:46]: Check if the frontdoor resource exists here, if it does, delete it and re-deploy.

File: Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1 [hunk 2]
```diff
@@ -202,38 +190,41 @@ function New-CmAzPaasWeb {
...
-					else {
-						Write-Verbose "Webapp $($appname) not found "
-						break
-					}
-					if (!$backEndPool.backendHostHeader) {
-						$backendHostHeader = ""
+				foreach ($serviceplan in $SettingsObject.appServicePlans) {
+					foreach ($app in $serviceplan.webapps) {
+						if($app.backendPool -match $backEndPool.Name){
+							$backEndDomainName = (Get-AzWebApp -ResourceGroupName $env:nameResourceGroup -Name  $app.generatedName).DefaultHostName
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1:204]: You can probably invert the if statement and add a continue, with the rest of the code being moved below. Which will mean reduce indentation abit.

File: Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1 [hunk 3]
```diff
@@ -43,15 +43,8 @@ function New-CmAzPaasWeb {
...
-			if (!$resourceGroup) {
-				$env:nameResourceGroup = Get-CmAzResourceName -Resource "ResourceGroup" -Architecture "PaaS" -Region $SettingsObject.location -Name $SettingsObject.resourceGroupName
-				Write-Verbose "Resourcegroup with tag $($SettingsObject.ResourceGroupTag) not found. New Resource Group '$($env:nameResourceGroup)' will be created."
-				New-AzResourceGroup -ResourceGroupName $env:nameResourceGroup -Location $SettingsObject.location -Force
-			}
-			else {
-				$env:nameResourceGroup = $resourceGroup.ResourceGroupName
-			}
+			$env:nameResourceGroup = Get-CmAzResourceName -Resource "ResourceGroup" -Architecture "PaaS" -Region $SettingsObject.location -Name $SettingsObject.resourceGroupName
+			New-AzResourceGroup -ResourceGroupName $env:nameResourceGroup -Location $SettingsObject.location -Force
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1:47]: Check if the frontdoor resource exists here, if it does, delete it and re-deploy.

File: Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1 [hunk 4]
```diff
@@ -202,38 +190,41 @@ function New-CmAzPaasWeb {
...
-					elseif ($resourceType -eq "Microsoft.Web/sites") {
-						$backEndDomainName = (Get-AzWebApp -ResourceGroupName $env:nameResourceGroup -Name  $appname).DefaultHostName
-					}
-					else {
-						Write-Verbose "Webapp $($appname) not found "
-						break
-					}
-					if (!$backEndPool.backendHostHeader) {
-						$backendHostHeader = ""
+				foreach ($serviceplan in $SettingsObject.appServicePlans) {
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1:201]: If we are not doing anything with the app service plan, we may be able to use "selectmany" to iterate over the webapps.

More info here: https://stackoverflow.com/questions/5241441/powershell-equivalent-of-linq-selectmany-method

File: Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1 [hunk 5]
```diff
@@ -202,38 +190,41 @@ function New-CmAzPaasWeb {
...
+								$backendHostHeader = $backEndDomainName
+							}
+							$backEndObject = New-AzFrontDoorBackendObject -Address $backEndDomainName -BackendHostHeader $backendHostHeader
+							$backendObjectArray.Add($backEndObject) > $null
+						}
 					}
-					else {
-						$backendHostHeader = $backEndDomainName
+				}
+				foreach ($apims in $SettingsObject.ApiManagementServices){
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1:216]: Can use the same "selectmany" approach here too

File: Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1 (hunk 2 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1]: Done

File: Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1 (hunk 5 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1]: Done

File: Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1 (hunk 4 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1]: Done

File: Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1 (hunk 3 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1:47]: > Check if the frontdoor resource exists here, if it does, delete it and re-deploy.

Done

File: Cloudmarque.Azure/Resources/Project/web.yml [hunk 6]
```diff
@@ -1,20 +1,24 @@
-resourceGroupName: # Required to create new resource group
-resourceGrouptag:  # provide a service tag for exisiting or new resource group
+﻿resourceGroupName: # Required to create new resource group
```
Comment (tm-luke) [inline Cloudmarque.Azure/Resources/Project/web.yml:1]: Can you tab out the comments in here and align them on the right.

File: Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1 [hunk 7]
```diff
@@ -202,31 +196,28 @@ function New-CmAzPaasWeb {
...
 					}
 					else {
-						Write-Verbose "Webapp $($appname) not found "
-						break
+						$backendHostHeader = $backEndDomainName
 					}
-					if (!$backEndPool.backendHostHeader) {
+					$backEndObject = New-AzFrontDoorBackendObject -Address $backEndDomainName -BackendHostHeader $backendHostHeader
+					$backendObjectArray.Add($backEndObject) > $null
+				}
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1:217]: Can you space out the loops a little

File: Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1 [hunk 8]
```diff
@@ -202,31 +196,28 @@ function New-CmAzPaasWeb {
...
+				$SettingsObject.ApiManagementServices | Where-Object { $_.backendPool -eq $backEndPool.Name } |ForEach-Object {
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/PaaS/web/New-CmAzPaasWeb.ps1:218]: Need a space between | and foreach-object

**Pull request merged**